### ISSUE TITLE
RasterLayer disable interpolation or antialiasing

### ISSUE THREAD

**Issue by abhishekBansal** [mapbox/mapbox-gl-native#15529]
Hello I am just starting out with Mapbox. I was playing with `ImageSourceTimeLapseActivity` in Android demo repository. I wish to turn anti aliasing or resampling off on the raster layers being rendered on map in this demo. After digging into `RasterLayer` code in mapbox-gl-native I tried following
 
```java
style.addLayer(layer.withProperties( new PropertyValue<>(layer.getRasterResampling().name, "nearest")));
```

and 

```java
// Add layer
        RasterLayer layer = new RasterLayer(ID_IMAGE_LAYER, ID_IMAGE_SOURCE);
        layer.setProperties( new PropertyValue<>(layer.getRasterResampling().name, "nearest"));

        style.addLayer(layer);
```
Both of these doesn't seem to work for me. I want my image to scale as is without any antialiasing or resampling. This is something I found is not possible with google maps. Is it possible with Mapbox at all?

**Comment by andrewharvey** [comment]
According to https://docs.mapbox.com/mapbox-gl-js/style-spec/#paint-raster-raster-resampling it is supported.

**Comment by abhishekBansal** [comment]
I found the solution, here is the code in SDK which is culprit
in `setProperties` method
```java
if (property instanceof PaintPropertyValue) {
        nativeSetPaintProperty(property.name, converted);
      } else {
        nativeSetLayoutProperty(property.name, converted);
      }
```
This `PaintPropertyValue` is an internal class of SDK and not available to clients directly. Upon more investogation I found `PropertyFactory.rasterResampling()` method in SDK which returns an instance of `PaintPropertyValue` and it worked.
Though this API design is little confusing my issue is now resolved. Thanks for the great work!
